energy-integration/fees-collector/src/lib.rs

```diff
@@ -64,8 +70,8 @@ pub trait FeesCollector:
                 let locked_rewards = self.lock_virtual(
                     self.get_base_token_id(),
                     reward.amount,
-                    caller.clone(),
-                    caller.clone(),
+                    orig_caller.clone(),
```

Comment:
Must send tokens to caller.
```suggestion
                    caller.clone(),
```

Reply:
Done.